1436790/1438462: Part 19: 20 files with MSDN link fixes.

DennisLee-DennisLee opened the pull request:
Contributes to PR #[2037](https://github.com/dotnet/docs/issues/2037) and applies to the following work items:
- [User Story 1436790: Content Update: update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1436790)
- [Task 1438462: Update MSDN links for .NET content](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1438462)

DennisLee-DennisLee committed:
Another 20 files with MSDN code fixes.

DennisLee-DennisLee commented:
The last modified file (**docs/framework/winforms/controls/customize-cells-and-columns-in-the-datagrid-by-extending-behavior.md**) contains the required changes to a paragraph that appears verbatim in 62 other files. Please inspect that one closely.

mairaw commented on docs/framework/network-programming/network-programming-samples.md:
```diff
@@ -10,40 +10,40 @@ This section contains descriptions and links to downloadable network programming
 
 ## In This Section  
- [Download Progress Indicator Technology Sample](https://msdn.microsoft.com/library/t8w6294a(v=vs.85).aspx)  
+ [Download Progress Indicator Technology Sample](https://docs.microsoft.com/previous-versions/dotnet/netframework-3.0/t8w6294a(v=vs.85))  
```
It seems that the latest version of this would be under the vs.90. Can you review the links on this page to see if you can find them under the vs.90?

```suggestion
 [Download Progress Indicator Technology Sample](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2008/t8w6294a(v=vs.90))  
```

DennisLee-DennisLee replied:
@mairaw I found two differences in the links inside those pages:

1. In the vs.85 version, the **System.Net** link in the See Also section goes to a current (non-previous-version) page on docs.microsoft.com containing long lists of classes/interfaces/enums/delegates, while the vs.90 version has the link going to an MSDN page (https://msdn.microsoft.com/en-us/library/btdf6a7e) containing just one class and one interface.

2. The **How to: Set Sample Settings** link goes to a more verbose topic from the vs.85 page than from the vs.90 page, though not as much as with **System.Net**. The article linked to from the vs.90 page basically says to follow what's described in VS2005/.NET2.0 and later versions. The explicit procedure in the linked articles are identical, so I think the pages are equivalent.

Also, the vs.85 and vs.90 pages themselves are the same other than some formatting differences, so I'd recommend sticking with the vs.85 page. (The previous-versions docset for vs.85 refers to dotnet as opposed to visualstudio for vs.90, so I assumed you'd prefer a dotnet URL if all other things were equal.)

mairaw replied:
Ok sounds good. Thanks for your analysis @DennisLee-DennisLee!

mairaw commented on docs/framework/performance/index.md:
```diff
@@ -41,10 +41,10 @@ If you want to create apps with great performance, you should design and plan fo
-|Windows Presentation Foundation (WPF)|[WPF Performance Suite](https://msdn.microsoft.com/library/67cafaad-57ad-4ecb-9c08-57fac144393e)|  
-|Silverlight|[Performance tips](https://msdn.microsoft.com/library/cc189071\(v=vs.95\).aspx)|  
-|ASP.NET|[ASP.NET Performance Overview](https://msdn.microsoft.com/library/f882bf1b-a009-4312-ac06-74370ffabc0b)|  
+|Windows Phone|[App performance considerations for Windows Phone 8](https://docs.microsoft.com/previous-versions/windows/apps/ff967560(v=vs.105))<br /><br /> [Windows Phone Application Analysis for Windows Phone 8](https://docs.microsoft.com/previous-versions/windows/apps/hh202934(v=vs.105))<br /><br /> [Get Your Windows Phone Applications in the Marketplace Faster](https://msdn.microsoft.com/mag ...
```
Let's remove this row about Windows Phone

```suggestion
```

mairaw commented on docs/framework/performance/index.md:
```diff
@@ -41,10 +41,10 @@ If you want to create apps with great performance, you should design and plan fo
-|ASP.NET|[ASP.NET Performance Overview](https://msdn.microsoft.com/library/f882bf1b-a009-4312-ac06-74370ffabc0b)|  
+|Windows Phone|[App performance considerations for Windows Phone 8](https://docs.microsoft.com/previous-versions/windows/apps/ff967560(v=vs.105))<br /><br /> [Windows Phone Application Analysis for Windows Phone 8](https://docs.microsoft.com/previous-versions/windows/apps/hh202934(v=vs.105))<br /><br /> [Get Your Windows Phone Applications in the Marketplace Faster](https://msdn.microsoft.com/mag ...
+|Windows Presentation Foundation (WPF)|[WPF Performance Suite](https://docs.microsoft.com/previous-versions/dotnet/netframework-4.0/aa969767(v=vs.100))|  
+|Silverlight|[Performance tips](https://docs.microsoft.com/previous-versions/windows/silverlight/dotnet-windows-silverlight/cc189071(v=vs.95))|  
```
same here with Silverlight

```suggestion
```

mairaw commented on docs/framework/performance/writing-large-responsive-apps.md:
```diff
@@ -31,7 +31,7 @@ This article provides tips for improving the performance of large .NET Framework
   
 ### Fact 3: Good tools make all the difference  
- Good tools let you drill quickly into the biggest performance issues (CPU, memory, or disk) and help you locate the code that causes those bottlenecks. Microsoft ships a variety of performance tools such as [Visual Studio Profiler](/visualstudio/profiling/beginners-guide-to-performance-profiling), [Windows Phone Analysis Tool](https://msdn.microsoft.com/library/e67e3199-ea43-4d14-ab7e-f7f1926625 ...
+ Good tools let you drill quickly into the biggest performance issues (CPU, memory, or disk) and help you locate the code that causes those bottlenecks. Microsoft ships a variety of performance tools such as [Visual Studio Profiler](/visualstudio/profiling/beginners-guide-to-performance-profiling), [Windows Phone Analysis Tool](https://msdn.microsoft.com/magazine/hh781024.aspx), and [PerfView](ht ...
```
```suggestion
 Good tools let you drill quickly into the biggest performance issues (CPU, memory, or disk) and help you locate the code that causes those bottlenecks. Microsoft ships a variety of performance tools such as [Visual Studio Profiler](/visualstudio/profiling/beginners-guide-to-performance-profiling) and [PerfView](https://www.microsoft.com/download/details.aspx?id=28567). 
```

mairaw commented on docs/framework/performance/writing-large-responsive-apps.md:
```diff
@@ -460,9 +460,9 @@ class Compilation { /*...*/
+- [.NET Performance Tips](https://docs.microsoft.com/previous-versions/dotnet/articles/ms973839(v%3dmsdn.10))
 - [Windows Phone Performance Analysis Tool](https://msdn.microsoft.com/magazine/hh781024.aspx)
-- [Find Application Bottlenecks with Visual Studio Profiler](https://msdn.microsoft.com/magazine/cc337887.aspx)
+- [Find Application Bottlenecks with Visual Studio Profiler](http://download.microsoft.com/download/3/A/7/3A7FA450-1F33-41F7-9E6D-3AA95B5A6AEA/MSDNMagazineMarch2008en-us.chm)
```
```suggestion
```

mairaw commented on docs/framework/performance/performance-tips.md:
```diff
@@ -38,6 +38,6 @@ The term *performance* generally refers to the execution speed of a program. You
 ## See also
 - [Performance](../../../docs/framework/performance/index.md)
-- [Programming Concepts](https://msdn.microsoft.com/library/65c12cca-af4f-4017-886e-2dbc00a189d6)
+- [Programming Concepts](https://docs.microsoft.com/previous-versions/visualstudio/visual-studio-2013/dd460655(v=vs.120))
```
```suggestion
```

mairaw commented on docs/framework/winforms/controls/customize-cells-and-columns-in-the-datagrid-by-extending-behavior.md:
```diff
@@ -62,7 +62,7 @@ The <xref:System.Windows.Forms.DataGridView> control provides a number of ways t
 -   References to the System, System.Windows.Forms, and System.Drawing assemblies.  
   
- For information about building this example from the command line for Visual Basic or Visual C#, see [Building from the Command Line](~/docs/visual-basic/reference/command-line-compiler/building-from-the-command-line.md) or [Command-line Building With csc.exe](~/docs/csharp/language-reference/compiler-options/command-line-building-with-csc-exe.md). You can also build this example in Visual Studi ...
+ For information about building this example from the command line for Visual Basic or Visual C#, see [Building from the Command Line](../../../visual-basic/reference/command-line-compiler/building-from-the-command-line.md) or [Command-line Building With csc.exe](../../../csharp/language-reference/compiler-options/command-line-building-with-csc-exe.md). You can also build this example in Visual S ...
```
I'd remove this last sentence where it appears. The docs site and the instructions on VS have changed since.

```suggestion
 For information about building this example from the command line for Visual Basic or Visual C#, see [Building from the Command Line](../../../visual-basic/reference/command-line-compiler/building-from-the-command-line.md) or [Command-line Building With csc.exe](../../../csharp/language-reference/compiler-options/command-line-building-with-csc-exe.md). You can also build this example in Visual Studio by pasting the code into a new project.
```

mairaw reviewed (changes requested):
Thank you @DennisLee-DennisLee. A few comments and also responded to your request to analyze the last one carefully.

mairaw committed:
Apply suggestions from code review

Added all but one of the autocommittable suggestions from Maira. (The other one I'll address in a separate commit.)

Co-Authored-By: DennisLee-DennisLee <v-dele@microsoft.com>
